gramps/plugins/gramplet/todo.py

```diff
@@ -184,13 +185,20 @@ def new_clicked(self, obj):
         """
         Create a new To Do note.
         """
-        from gramps.gui.editors import EditNote
-        note = Note()
-        note.set_type(NoteType.TODO)
-        try:
-            EditNote(self.gui.dbstate, self.gui.uistate, [], note, self.created)
-        except AttributeError:
-            pass
+        nav_type = self.uistate.viewmanager.active_page.navigation_type()
+        active_handle = self.get_active(nav_type)
+        if active_handle:
+            from gramps.gui.editors import EditNote
+            note = Note()
+            note.set_type(NoteType.TODO)
+            try:
+                EditNote(self.gui.dbstate, self.gui.uistate, [], note, self.created)
+            except AttributeError:
+                pass
+        else:
+            WarningDialog(_("No active object"),
+                          _("No active object") + _(":") + _(nav_type),
```

Comment:
The string "No active object" appears in the title bar, primary message and secondary message.  Can we think of something more helpful?  For example:  "First select the object to which you want to attach a note".

Reply:
I changed the message as you said.

Reply:
New Example message

![example message](https://user-images.githubusercontent.com/8924713/49119044-db500380-f2fa-11e8-8259-211db733d887.png)

Reply:
@Nick-Hall Is this message ok to you?